### PR TITLE
Add unit and e2e tests for promocode service

### PR DESCRIPTION
**Pull request by irnrogalik**

### CODE REVIEW

File: src/modules/promocode/dto/PromocodeAddDto.ts [hunk 1]
```diff
@@ -23,8 +23,8 @@ export class PromocodeAddDto implements IAddPromoCode {
     isOneTime = false;
 
     @ApiPropertyOptional({ default: null })
-    startDate: Date = null;
+    startDate?: Date = null;
```
Comment (dianager) [inline src/modules/promocode/dto/PromocodeAddDto.ts:26]: in promocode microservice this field is not required https://github.com/irnrogalik/promo_microservice/pull/3/files#diff-ffeaaabb9ffd80650b1a92b8021008d8407002f18f10d09b6177a81f24287180R7. Data types should be consistent

File: src/modules/promocode/promo.interface.ts [hunk 2]
```diff
@@ -35,8 +35,8 @@ export interface IAddPromoCode {
     name: string;
     percent: number;
     isOneTime: boolean;
-    startDate: string | Date;
-    endDate: string | Date;
+    startDate?: string | Date;
```
Comment (dianager) [inline src/modules/promocode/promo.interface.ts:38]: the same comment

File: src/modules/promocode/promo.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    pageOptions,
+    promocode,
+    promocodeName,
+    promocodeToAdd,
+    promocodeToRemove,
+} from './promo.fixture';
+import { MockPromoCodeService } from './promo.mock.service';
+import { PromocodeService } from './promo.service';
+
+class ClientGrpcMock {
```
Comment (dianager) [inline src/modules/promocode/promo.service.spec.ts]: nit: can be also moved to mock file

Comment (irnrogalik) [reply src/modules/promocode/promo.service.spec.ts]: @dianager  I have a limit on the number of classes in a file. The max-classes-per-file setting is currently set to 1. Should I change it or move ClientGrpcMock to a different file, or left as is?

Comment (dianager) [reply src/modules/promocode/promo.service.spec.ts]: I would better move to promo.client.mock.ts file, in case in future you will need it in other tests

**Review by dianager — APPROVED** [review]

**Pull request merged**